tasks/monitoring.yml

```diff
@@ -1,61 +1,74 @@
 
-- block:
-  - name: Create temp directory
-    tempfile:
-      path: /tmp
-      state: directory
-      suffix: cloud_ops_shell_scripts
-    register: tempfolder
+- name: Create temp directory
+  tempfile:
+    path: /tmp
+    state: directory
+    suffix: cloud_ops_shell_scripts
+  register: tempfolder
+  check_mode: no
+  changed_when: false
 
-  - name: Download script
-    get_url:
-      url: https://dl.google.com/cloudagents/add-monitoring-agent-repo.sh
-      dest: "{{ tempfolder.path }}/add-monitoring-agent-repo.sh"
-      force: yes
-      mode: 0755
+- name: Download script
+  get_url:
+    url: https://dl.google.com/cloudagents/add-monitoring-agent-repo.sh
+    dest: "{{ tempfolder.path }}/add-monitoring-agent-repo.sh"
+    mode: 0755
+  check_mode: no
+  changed_when: false
 
+- when: package_state == 'present'
+  block:
   - name: Add repo and install agent
     command:
       chdir: "{{ tempfolder.path }}"
-      cmd: "bash add-monitoring-agent-repo.sh --also-install --version={{ version }}"
+      cmd: "bash add-monitoring-agent-repo.sh --also-install --version={{ version }} {{ '--dry-run' if ansible_check_mode else '' }}"
     environment:
       REPO_CODENAME: "{{ ansible_distribution_release }}"
     register: result
+    check_mode: no
     changed_when: "'No changes made.' not in result.stdout_lines"
     notify: restart monitoring agent
 
-  when: package_state == "present"
+  - name: Copy main config file onto the remote machine
+    copy:
+      src: "{{ main_config_file }}"
+      dest: /etc/stackdriver/collectd.conf
+      force: yes
+      mode: 0644
+      validate: '/opt/stackdriver/collectd/sbin/stackdriver-collectd -tC %s'
+    when: main_config_file|length > 0
+    notify: restart monitoring agent
 
-- name: Uninstall agent
-  package:
-    name: "{{ monitoring_package_name }}"
-    state: "{{ package_state }}"
-  when: package_state == "absent"
-#TODO(rmoriar1): Clean up the repo when package_state is absent.
+  - name: Copy additional configs onto the remote machine
+    copy:
+      src: "{{ item }}"
+      dest: /etc/stackdriver/collectd.d/
+      force: yes
+      mode: 0644
+      validate: '/opt/stackdriver/collectd/sbin/stackdriver-collectd -tC %s'
+    with_fileglob:
+      - "{{ additional_config_dir }}/*.conf"
+    when: additional_config_dir|length > 0
+    notify: restart monitoring agent
 
-- name: Copy main config file onto the remote machine
-  copy:
-    src: "{{ main_config_file }}"
-    dest: /etc/stackdriver/collectd.conf
-    force: yes
-    mode: 0644
-    validate: "{{ monitoring_binary }} -tC %s"
-  when: main_config_file|length > 0
-  notify: restart monitoring agent
+- when: package_state == 'absent'
+  block:
+  - name: Uninstall agent and remove repo
+    command:
+      chdir: "{{ tempfolder.path }}"
+      cmd: "bash add-monitoring-agent-repo.sh --uninstall --remove_repo {{ '--dry-run' if ansible_check_mode else '' }}"
```

Comment:
Combine install and uninstall into a single script invocation? That way you avoid always having a skipped task.

Reply:
Done.